### ISSUE TITLE
[test][mono][linux][x64] System.IO.Tests.FileStream_DisposeAsync.DisposeAsyncFlushes library test failed 

### ISSUE THREAD

**Issue by MaximLipnin** [dotnet/runtime#65835]
The `System.IO.Tests.FileStream_DisposeAsync.DisposeAsyncFlushes` library test failed on the following CI jobs when running on the rolling build ([Build](https://dev.azure.com/dnceng/public/_build/results?buildId=1629951&view=ms.vss-test-web.build-test-results-tab&runId=45214896&paneView=dotnet-dnceng.dnceng-build-release-tasks.helix-test-information-tab&resultId=151978), [Console log](https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-main-e2e8fbc0f9e546ac9d/System.IO.FileSystem.Tests/1/console.3b560bfa.log?sv=2019-07-07&se=2022-03-16T08%3A52%3A25Z&sr=c&sp=rl&sig=%2F9FxDZoxrjgB2%2FaoaweLTZxwwdhlJvtxVcCtaJmkyZA%3D)):
- `net7.0-Linux-Release-x64-Mono_release-(Debian.10.Amd64.Open)`
- `net7.0-Linux-Release-x64-Mono_release-RedHat.7.Amd64.Open`
- `net7.0-Linux-Release-x64-Mono_release-(Centos.7.Amd64.Open)`

with the message:
```
    System.IO.Tests.FileStream_DisposeAsync.DisposeAsyncFlushes [FAIL]
      Assert.Equal() Failure
      Expected: 100
      Actual:   200
      Stack Trace:
        /_/src/libraries/System.IO.FileSystem/tests/FileStream/DisposeAsync.cs(39,0): at System.IO.Tests.FileStream_DisposeAsync.DisposeAsyncFlushes()
        --- End of stack trace from previous location ---
```
Upd. Same for window legs (https://github.com/dotnet/runtime/issues/65835#issuecomment-1049795562, https://github.com/dotnet/runtime/issues/65835#issuecomment-1050576596)

**Comment by rzikm** [comment]
Same issue has been obserbed on Windows as well: https://dev.azure.com/dnceng/public/_build/results?buildId=1628534&view=logs&j=457f7e88-dfa2-5bd9-f871-fdf124c2477d&t=bfe52dfb-2099-5c7f-ee52-70a1d81c544e&l=358

**Comment by MaximLipnin** [comment]
Another example of the failure on Windows - https://github.com/dotnet/runtime/issues/65872

**Comment by adamsitnik** [comment]
Fixed by #65899
